Emerald Insight.js

```diff
@@ -150,8 +150,16 @@ function scrape(doc, url) {
 			// trying to correct for this
 			for (let i = 0; i < item.creators.length; i++) {
 				if (!item.creators[i].firstName && item.creators[i].lastName.includes(" ")) {
-					item.creators[i].firstName = item.creators[i].lastName.match(/^\w+\s+(.+)/)[1];
-					item.creators[i].lastName = item.creators[i].lastName.replace(/\s.+/, "");
+					var author = doc.querySelector("a[href*='" + item.creators[i].lastName.split(" ")[0] + "'][class='contrib-search']");
+					if(author) {
+						item.creators[i].firstName = text(author, "span[class=given-names]");
+						item.creators[i].lastName = text(author, "span[class=surname]");
+					}
+					else {
+						Z.debug("No span tag for " + item.creators[i].lastName);
+						item.creators[i].firstName = item.creators[i].lastName.split(" ")[0];
```

Comment:
You switch here to the `firstName`, which should only been done for the editors. Can we make an additional condition and do the splitting differently for authors and editors? Moreover, I suggest to make a comment here somewhere, such that we remember also later.

Reply:
> Yeah, I noticed that. This is a mess, but we can assume this is just for `AU` until we have evidence to the contrary.

I thought @dstillman previous comment implied that only authors would be lastname firstname and for others it would be firstname lastname every time, but I may have misunderstood

Reply:
You are going through the `item.creators` list and there are the authors as well as the editors. However, for each of them you have the `creatorType` and therefore you can decide if it is an author or editor.

Reply:
Ok I can add an else if in case there was no span info but the creator is still an author. I just assumed that only authors have their info in the span embedded, so that they will always be included in the first `if`